src/app/ReadClient.cpp

```diff
@@ -1102,10 +1108,33 @@ void ReadClient::OnResubscribeTimerCallback(System::Layer * apSystemLayer, void
         {
             _this->mReadPrepareParams.mSessionHolder->AsSecureSession()->MarkAsDefunct();
         }
```

Comment:
With this PR, we reattempt CASE below if the session is defunct no matter what.  Given that, should we just unconditionally mark the session defunct in `ScheduleResubscription`?  Noting that the session manager will move `mReadPrepareParams.mSessionHolder` to a newer / better session if one appears, there seems to be a subtle race condition here where, if a new session was established in the interim between `ScheduleResubscription` and `OnResubscribeTimerCallback`, we will now be marking an incorrect, newer session defunct.

I am no longer following what consideration of the `mForceCaseOnNextResub` flag in `OnResubscribeTimerCallback` does for us except to break the callback loop with `_this->Close(err, allowResubscribeOnError=false);`.  But doesn't this method have enough context to do that anyway without the flag?